P5/Source/Specs/schemaSpec.xml

```diff
@@ -45,28 +45,38 @@ $Id$
+    <desc xml:lang="en" versionDate="2021-02-02">enforces the
+    deprecation of <gi>schemaSpec</gi> as a child of anything other
+    than <gi>encodingDesc</gi>, <gi>front</gi>, <gi>body</gi>,
     <gi>back</gi>, <gi>div</gi>, or any of the numbered division
-    elements; it should be removed when the class memberships and
-    content models are altered so that this restriction is enforced by
-    closed schema.</desc>
+    elements. <!-- this <constraintSpec> should be removed when the
+    class memberships and content models are altered so that this
+    restriction is enforced by closed schema, which is expected to
+    occur in October 2021. --></desc>
+    <desc type="deprecationInfo">
+      The use of <gi>schemaSpec</gi> as a child of any element other
+      than <gi>encodingDesc</gi>, <gi>front</gi>, <gi>body</gi>,
+      <gi>back</gi>, <gi>div</gi>, or one of the numbered division
+      elements is being withdrawn.</desc>
     <constraint>
-      <sch:assert test="  parent::teix:egXML
-                        | parent::tei:encodingDesc
-                        | parent::tei:front
-                        | parent::tei:back
-                        | parent::tei:body
-                        | parent::tei:div
-                        | parent::tei:div1
-                        | parent::tei:div2
-                        | parent::tei:div3
-                        | parent::tei:div4
-                        | parent::tei:div5
-                        | parent::tei:div6
-                        | parent::tei:div7">WARNING: use of deprecated construct — the schemaSpec element will no longer be a valid child of <sch:value-of
-                        select="name(..)"/> as of 2021-10-23; instead, it should be a child of “encodingDesc”, “front”, “body”, “back”, or a division element.</sch:assert>
+      <sch:rule context="tei:schemaSpec|teix:schemaSpec">
+	<sch:assert test="             parent::teix:egXML
+			  | parent::tei:encodingDesc  | parent::teix:encodingDesc
+			  | parent::tei:front         | parent::teix:front
+			  | parent::tei:back          | parent::teix:back
+			  | parent::tei:body          | parent::teix:body
+			  | parent::tei:div           | parent::teix:div
+			  | parent::tei:div1          | parent::teix:div1
+			  | parent::tei:div2          | parent::teix:div2
+			  | parent::tei:div3          | parent::teix:div3
+			  | parent::tei:div4          | parent::teix:div4
+			  | parent::tei:div5          | parent::teix:div5
+			  | parent::tei:div6          | parent::teix:div6
+			  | parent::tei:div7          | parent::teix:div7"
+		    role="nonfatal">WARNING: use of deprecated construct — the “schemaSpec” element will no longer be a valid child of “<sch:value-of
+                        select="name(..)"/>” as of 2021-10-23; instead, it should be a child of “encodingDesc”, “front”, “body”, “back”, or a division element.</sch:assert>
```

Comment:
I wonder if the date should be changed to February 2022?

Reply:
Only if we defer to 4.3. From the beginning this was supposed to be a short (6–12 month) deprecation period.

Reply:
ah, ok.